CRM/Financial/BAO/FinancialAccount.php

```diff
@@ -361,12 +361,15 @@ public static function validateFinancialAccount($financialAccountId, $financialA
    * with Financial Account
    *
    * @param array $params
+   *   Holds submitted formvalues and params from api for updating/adding contribution
    *
    * @param int $contributionID
+   *   Contribution ID
    *
    * @param obj $form
+   *   Holds form object of CRM_Contribute_Form_Contribution
    *
-   * @return string
+   * @return bool
    *
    */
   public static function checkFinancialTypeHasDeferred($params, $contributionID = NULL, $form = NULL) {
```

Comment:
I don't think it is good style to pass $form object to BAO. And I don't see where $form is used in function. Can we remove it from call signature?